Firestore/core/src/api/settings.h

```diff
@@ -110,56 +109,50 @@ class LocalCacheSettings {
   explicit LocalCacheSettings(Kind kind) : kind_(std::move(kind)) {
   }
   Kind kind_;
-
- private:
-  friend class Settings;
 };
 
 class PersistentCacheSettings : public LocalCacheSettings {
+  friend class Settings;
+
  public:
   PersistentCacheSettings()
       : LocalCacheSettings(LocalCacheSettings::Kind::kPersistent),
         size_bytes_(Settings::DefaultCacheSizeBytes) {
   }
   PersistentCacheSettings WithSizeBytes(int64_t size) const;
 
-  friend bool operator==(const PersistentCacheSettings& lhs,
-                         const PersistentCacheSettings& rhs) {
-    return lhs.kind_ == rhs.kind_ && lhs.size_bytes_ == rhs.size_bytes_;
+  int64_t size_bytes() const {
+    return size_bytes_;
   }
 
-  friend bool operator!=(const PersistentCacheSettings& lhs,
-                         const PersistentCacheSettings& rhs) {
-    return !(lhs == rhs);
-  }
   size_t Hash() const override;
 
  private:
-  friend class Settings;
-
   int64_t size_bytes_;
 };
 
 class MemoryCacheSettings : public LocalCacheSettings {
+  friend class Settings;
+
  public:
   MemoryCacheSettings()
       : LocalCacheSettings(LocalCacheSettings::Kind::kMemory) {
   }
-  friend bool operator==(const MemoryCacheSettings& lhs,
-                         const MemoryCacheSettings& rhs) {
-    return lhs.kind_ == rhs.kind_;
-  }
-
-  friend bool operator!=(const MemoryCacheSettings& lhs,
-                         const MemoryCacheSettings& rhs) {
-    return !(lhs == rhs);
-  }
   size_t Hash() const override;
-
- private:
-  friend class Settings;
 };
 
+bool operator!=(const Settings& lhs, const Settings& rhs);
```

Comment:
```
bool operator==(const Settings& lhs, const Settings& rhs);
```
is missing.

Reply:
This is defined in class declaration at line 79.